File: iosApp/iosApp/BackendConfigRepository.swift
import Foundation
import FirebaseFirestore

class BackendConfigRepository {
    private let db = Firestore.firestore()

    func getFormFrequencyMonths() async throws -> Int {
        let doc = try await db.collection("config").document("formFrequency").getDocument()
        return doc.data()?["frequencyMonths"] as? Int ?? 1
    }
}

File: composeApp/src/androidMain/kotlin/com/humblesolutions/indsphinx/repository/BackendConfigRepository.kt
package com.humblesolutions.indsphinx.repository

import com.google.firebase.firestore.FirebaseFirestore
import kotlinx.coroutines.tasks.await

class BackendConfigRepository : ConfigRepository {
    private val db = FirebaseFirestore.getInstance()

    override suspend fun getFormFrequencyMonths(): Int {
        val doc = db.collection("config").document("formFrequency").get().await()
        return doc.getLong("frequencyMonths")?.toInt() ?: 1
    }
}
